hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3223,7 +3223,7 @@ func (r *HostedClusterReconciler) validateConfigAndClusterCapabilities(ctx conte
 func (r *HostedClusterReconciler) validateReleaseImage(ctx context.Context, hc *hyperv1.HostedCluster) error {
 	var pullSecret corev1.Secret
 	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: hc.Namespace, Name: hc.Spec.PullSecret.Name}, &pullSecret); err != nil {
-		return fmt.Errorf("failed to get pull secret: %w", err)
+		return apierrors.NewNotFound(appsv1.Resource("secrets"), err.Error())
```

Comment:
you could just `return err`. In any case no change is needed. This is working as intended, otherwise the unit test you added would fail.